command/node_status.go

```diff
@@ -160,3 +164,27 @@ func (c *NodeStatusCommand) Run(args []string) int {
 	}
 	return 0
 }
+
```

Comment:
I would remove both these methods and replace them with one that takes the original nodes and returns a map of their original ids to their shortened ones.

`shortIds(nodes []*api.NodeListStub) map[string]string`

With this method I think you could be a bit more advanced with how you shorted. You could start at 8 characters and if there is a collision, then you up it to 9, etc, until you get no collisions. Further it is probably still worth while to shorten the non-custom ones.